### PR TITLE
added var for maxLengthForImgRequest

### PR DESCRIPTION
**Pull request by liebhaeuser**
We just had the issue that a GET request sent in the `sendBeacon` function raised an alert in a security filter because the parameters were not encoded.

For what reason do you prefer a GET request over a POST for urls with a length shorter than 2000? 

It would be nice to have the maxLengthForImgRequest as global variable, e.g. to be able to set it to 0 to always send a POST request if the XMLHttpRequest API is supported by the browser.

### CODE REVIEW

File: lib/vars.js [hunk 1]
```diff
@@ -200,7 +203,8 @@ const defaultVars: {
   page: undefined,
   wrapEventHandlers: false,
   wrappedEventHandlersOriginalFunctionStorageKey: '__weaselOriginalFunctions__',
-  wrapTimers: false
+  wrapTimers: false,
+  maxLengthForImgRequest: 2000
```
Comment (bripkens) [inline lib/vars.js]: Could you please make this configurable via https://github.com/instana/weasel/blob/master/lib/commands.js#L68-L70 ?

File: lib/vars.js [hunk 2]
```diff
@@ -178,7 +178,10 @@ const defaultVars: {
   //
   // Set via:
   // eum('wrapTimers', true)
-  wrapTimers: boolean
+  wrapTimers: boolean,
+
+  //
```
Comment (bripkens) [inline lib/vars.js]: Please add documentation here similar to the other config values above that explains what it does and how to change it.

**Comment by bripkens** [comment]
This sounds unusual because that should not be possible according to [1]. Would you be able to provider a reproducer to justify this change?

[1] https://github.com/instana/weasel/blob/master/lib/transmission.js#L46

**Comment by liebhaeuser** [comment]
I come back to you as soon as I know more about the issue with the security filter. It's quite possible the something in the customers log chain does the decoding.

**Comment by liebhaeuser** [comment]
Could you please answer my question according to GET via POST?

> For what reason do you prefer a GET request over a POST for urls with a length shorter than 2000?

**Comment by bripkens** [comment]
> For what reason do you prefer a GET request over a POST for urls with a length shorter than 2000?

Image request execution is faster in the browser. We noticed this in one of our long polling apps that execution of XHR calls has a significant overhead. Though to be fair, this probably not a problem that the average website will be affected by.

**Comment by bripkens** [comment]
Closing this PR due to inactivity.

**Pull request closed without merging**